store/sqlstore/channel_store.go

```diff
@@ -93,11 +98,16 @@ func (db channelMemberWithSchemeRoles) ToModel() *model.ChannelMember {
 
 	// Identify any system-wide scheme derived roles that are in "Roles" field due to not yet being migrated,
 	// and exclude them from ExplicitRoles field.
+	schemeGuest := db.SchemeGuest.Valid && db.SchemeGuest.Bool
 	schemeUser := db.SchemeUser.Valid && db.SchemeUser.Bool
 	schemeAdmin := db.SchemeAdmin.Valid && db.SchemeAdmin.Bool
 	for _, role := range strings.Fields(db.Roles) {
 		isImplicit := false
-		if role == model.CHANNEL_USER_ROLE_ID {
+		if role == model.CHANNEL_GUEST_ROLE_ID {
+			// We have an implicit role via the system scheme. Override the "schemeUser" field to true.
```

Comment:
This code is for handling pre-advanced permissions users. I'm not sure it needs to be aware of the guest roles. Then again, it doesn't do any harm other than adding a few more lines of code.

Reply:
Yes, that was my point too... I was struggling with that concept all over the PR, but I think is more conservative to maintain that consistency because possible multiversion migration with a weird situation.

Reply:
Yeah, I think I'm on board with that approach - it feels "safer" on the whole, even though there's more code as a result.